moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -423,6 +423,12 @@ public final void setFailOnUnknown(boolean failOnUnknown) {
    */
   public abstract void skipValue() throws IOException;
 
+  /**
+   * Reads the next JSON name or value without decoding it. White space and comments will be
+   * removed.
```

Comment:
Should we be more formal here? Like should we say anything about which characters are encoded? Should we make promises around how numbers are formatted?